app/api/create-checkout-session/route.ts

```diff
@@ -0,0 +1,43 @@
+import { NextResponse } from "next/server";
+
+const SERVER_URL = process.env.SERVER_URL || "http://127.0.0.1:8000";
+
+export async function POST(request: Request) {
```

Comment:
I don't think we need this, can directly call the backend python server from frontend instead of going through nextjs backend

Reply:
Your right we dont really need this that much and it even kind of slows down performance since it creates a extra request, but I think we should keep it primarily for its proxy functionality. It sort of acts like a middleware it conceals the backend server's API route from direct client exposure. This just adds a layer of security and potentially helps mitigate possible ddos attacks.

Without the api route: the full URL of the backend API `(https://api.yourbackend.com/payment/create-checkout-session)` would be visible in the client-side code. So it would show the domain of the backend server and the endpoint structure

With api route: the frontend code only knows about `/api/create-checkout-session`. So whats only going to be visible is `/api/create-checkout-session`

Reply:
Interesting! OK we don't care about performance for this endpoint. We only have to weigh security (want more) and code complexity (want less).

I like that doing this doesn't expose backend endpoint, so let's do this. However, we need authenticate for nextjs backend endpoints. The current one doesn't seem to be auth'd.

I have auth'd the python server endpoint on my side.